go/adbc/pkg/snowflake/driver.go

```diff
@@ -97,7 +97,27 @@ func setErrWithDetails(err *C.struct_AdbcError, adbcError adbc.Error) {
 		return
 	}
 
+	for i := 0; i < 5; i++ {
+		err.sqlstate[i] = C.char(adbcError.SqlState[i])
+	}
+
 	if err.vendor_code != C.ADBC_ERROR_VENDOR_CODE_PRIVATE_DATA {
+		// Caller is not interested in `private_data` if `vendor_code` is not
+		// `ADBC_ERROR_VENDOR_CODE_PRIVATE_DATA` so let's use the field for
+		// `vendor_code` instead, but make sure it's not set to the value
+		// that would indicate `private_data` is set.
+		if adbcError.VendorCode != C.ADBC_ERROR_VENDOR_CODE_PRIVATE_DATA {
+			err.vendor_code = C.int(adbcError.VendorCode)
+		}
+		setErr(err, adbcError.Msg)
+		return
+	}
+
+	numDetails := len(adbcError.Details)
+	// If there are no details, but we have a `VendorCode`, let's override
+	// `vendor_code` and not populate `private_data` with the error details.
+	if numDetails == 0 && adbcError.VendorCode != 0 && adbcError.VendorCode != C.ADBC_ERROR_VENDOR_CODE_PRIVATE_DATA {
+		err.vendor_code = C.int(adbcError.VendorCode)
```

Comment:
@zeroshade if you are OK with this approach I will change the `go.tmpl` file and `make regenerate`.

Reply:
This looks good to me. I wonder if there's any good way we could add a test for this (perhaps using the sqlite driver or otherwise?)

Reply:
The SQLite driver is not written in Go, so I wouldn't be able to exercise this code with it. And the individual Go driver tests seem to skip the FFI layer.

Reply:
> So this is strictly an issue with the C Interop layer? There are interop tests in both the Rust and C# implementations that exercise the Snowflake driver.

Yes. Can you force a Snowflake error from the C# interop tests then assert on the vendor code?